### PR TITLE
Ddp 4592 add server-side recaptcha confirmation

### PR DESCRIPTION
**Pull request by MocanaAtBroad**
## Context

We wire in the server-side check for ReCaptcha.
We store the study key in the database in the umbrella study.
Updated tests for the route that checks invitation
Added the recaptcha site key to the study server.
The key is study specific and stored initially in vault. I set our current dev site key.

## Checklist

- [X ] I have labeled the type of changes involved using the `C-*` labels.
- [ X] I have assessed potential risks and labeled using the `R-*` labels.
- [ X] I have considered error handling and alerts, and added `L-*` labels as needed.
- [ x] I have considered security and privacy, and added `I-*` labels as needed
- [ x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [X ] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ X] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [X ] I have written automated positive tests
- [X ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [ ] These changes require no special release procedures--just code!
- [ X] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dto/StudyDto.java [hunk 1]
```diff
@@ -86,4 +89,12 @@ public String getStudyEmail() {
...
         return dataExportEnabled;
     }
+
+    public String getRecaptchaSiteKey() {
+        return recaptchaSiteKey;
+    }
+
+    public void setRecaptchaSiteKey(String key) {
+        this.recaptchaSiteKey = key;
+    }
```
Comment (MocanaAtBroad) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/db/dto/StudyDto.java:99]: Setter need to facilitate testing

File: pepper-apis/src/test/java/org/broadinstitute/ddp/route/InvitationCheckStatusRouteTest.java [hunk 2]
```diff
@@ -105,33 +123,34 @@ public void testCheckStatus_badInvitation() {
...
         InvitationDto invitation = TransactionWrapper.withTxn(handle -> handle.attach(InvitationFactory.class)
                 .createRecruitmentInvitation(testData.getStudyId(), "invite" + System.currentTimeMillis()));
         try {
             var payload = new InvitationCheckStatusPayload(
                     testData.getAuth0ClientId(),
-                    invitation.getInvitationGuid());
-            String url = RouteTestUtil.getTestingBaseUrl() + RouteConstants.API.INVITATION_CHECK;
+                    invitation.getInvitationGuid(), "mockTokenValue");
+            String url = testServer.baseUrl() + RouteConstants.API.INVITATION_CHECK;
             url = url.replace(RouteConstants.PathParam.STUDY_GUID, testData.getStudyGuid());
```
Comment (MocanaAtBroad) [inline pepper-apis/src/test/java/org/broadinstitute/ddp/route/InvitationCheckStatusRouteTest.java:154]: Here we use our test server with the one route.

File: pepper-apis/src/test/java/org/broadinstitute/ddp/util/StudyDataLoaderTest.java [hunk 3]
```diff
@@ -918,7 +918,7 @@ public void testLoadBloodConsent() throws Exception {
         UserDto userDto = new UserDto(pretendUserId, pretendAuth0UserId, pretendUserGuid, pretendUserGuid, null,
                 null, now, now);
         StudyDto studyDto = new StudyDto(pretendStudyId, pretendStudyGuid, "MBC", null, null,
-                1L, 2L, null, false, null, false);
+                1L, 2L, null, false, null, null, false);
```
Comment (MocanaAtBroad) [inline pepper-apis/src/test/java/org/broadinstitute/ddp/util/StudyDataLoaderTest.java:922]: More refactoring casualties.

File: pepper-apis/src/test/java/org/broadinstitute/ddp/util/TestServer.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    public TestServer stopServer() {
+        service.stop();
+        service.awaitStop();
+        return this;
+    }
+
+    public String baseUrl() {
+        return "http://localhost:" + PORT;
+    }
+}
```
Comment (MocanaAtBroad) [inline pepper-apis/src/test/java/org/broadinstitute/ddp/util/TestServer.java:36]: Not much but we can definitely reuse this.

**Review by ssettipalli — APPROVED** [review]
looks good to me

**Review by yufengwng — APPROVED** [review]
Nice!

**Pull request merged**